From cabe495f7512dd1d622c04b7137abcafe13113b5 Mon Sep 17 00:00:00 2001
From: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Date: Thu, 18 Apr 2013 08:10:04 +0000
Subject: [PATCH 550/858] qlcnic: Stop traffic before performing loopback test

Before conducting loopback test by sending packets, driver should stop transmit
queue and turn off carrier.

Signed-off-by: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Signed-off-by: Shahed Shaikh <shahed.shaikh@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit a4325ea242d9d0889cb7e86811fef3ff3d714d29)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
index cd5ae88..284d628 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.c
@@ -1500,6 +1500,12 @@ int qlcnic_83xx_loopback_test(struct net_device *netdev, u8 mode)
   }
  } while ((adapter->ahw->linkup && ahw->has_link_events) != 1);
 
+ /* Make sure carrier is off and queue is stopped during loopback */
+ if (netif_running(netdev)) {
+  netif_carrier_off(netdev);
+  netif_stop_queue(netdev);
+ }
+
  ret = qlcnic_do_lb_test(adapter, mode);
 
  qlcnic_83xx_clear_lb_mode(adapter, mode);
-- 
1.8.3.1